### DISCUSSION TITLE
MC Bedrock: Internet IP incorrect, `./gameserver monitor` fails, but server works internally and externally

### DISCUSSION THREAD

**Discussion by rjsears** [GameServerManagers/LinuxGSM#4689] [General]
As the title says, I am running an MC Bedrock server on an Ubuntu box. Install went smoothly (first time user to LinuxGSM v24.3.1), I am able to see the server at `https://mcsrvstat.us/` and my kids can login both with their PCs and  Xboxs and play with no problems. However when I run `./mcbserver detail ` I am getting an incorrect 'Internet IP' and when I run `./mcbserver monitor` it fails: 

```
Minecraft Bedrock Server Details
===========================
Server name:  Abby's MC Server
Server IP:    0.0.0.0:19132
Internet IP:  xx.xx.xx.66:19132.   <---- This is incorrect
Maxplayers:   10
Game mode:    survival
Status:       STARTED

mcbserver Script Details
===========================
Script name:       mcbserver
LinuxGSM version:  v24.3.1
glibc required:    2.27
User:              gameuser
Location:          /home/gameuser/gameserver
Config file:       /home/gameuser/gameserver/serverfiles/server.properties
```

```
gameuser@mc-server02 ~/gameserver$ ./mcbserver monitor
[  OK  ] Monitoring mcbserver: Checking session: OK
[ FAIL ] Monitoring mcbserver: Querying port: gsquery: 127.0.0.1:19132 : 4/1 : WAIT
[ FAIL ] Monitoring mcbserver: Querying port: gsquery: xx.xx.xx.66:19132 : 4/1 : WAIT  <--- Incorrect External IP
[ FAIL ] Monitoring mcbserver: Querying port: gsquery: 0.0.0.0:19132 : 4/1 : WAIT
```

Running `curl https://ipinfo.io/ip` returns the servers correct external IP address.

So while the server is working great and I can see it correctly using  `https://mcsrvstat.us/`, I would like to have the internal details and monitor work correctly as well.

Thank You

**Comment by theBeaz** [comment]
Seconding that I am having this issue with gsquery on a Bedrock server (except I'm not having the wrong external IP). It fails to monitor the server. It can also successfully be queried from https://mcsrvstat.us/

**Comment by theBeaz** [comment]
https://github.com/GameServerManagers/LinuxGSM/issues/4466

The workaround in this Issue fixed my problem. The issue was declared fixed, but doesn't appear to be the case.
